simple/ud_pingpong.c

```diff
@@ -441,7 +450,10 @@ static int server_connect(void)
 
 	do {
 		ret = fi_cq_read(rcq, &comp, 1);
-		if (ret < 0) {
+		if (ret == -FI_EAGAIN) {
+			ret = 0;
+			continue;
+		} else if (ret < 0) {
```

Comment:
There are several places where we check for ret == -FI_EAGAIN and modify ret = 0.  It would be better to modify these loops as well, so that we loop on ret == -FI_EAGAIN.